scripts/deploy.ts

```diff
@@ -1,122 +1,194 @@
-import { use } from 'chai';
 import { ethers } from 'hardhat';
+import { SignerWithAddress } from '@nomiclabs/hardhat-ethers/signers';
+import { BigNumber } from 'ethers';
 const hre = require('hardhat');
 const fs = require('fs');
 
-async function main() {
-  const output = {
-    walletImplementation: '',
-    walletFactory: '',
-    forwarderImplementation: '',
-    forwarderFactory: ''
-  };
-
-  const feeData = await ethers.provider.getFeeData();
-  const gasParams = {
-    gasPrice: feeData.gasPrice
-  };
+async function selfTransferForV1(
+  deployer: SignerWithAddress,
+  gasPrice: BigNumber | null,
+  totalTxnCount: number
+) {
+  const txCount = await deployer.getTransactionCount();
+  console.log('Tx Count: ', txCount);
+  const selfTransactions = totalTxnCount - txCount;
+  for (let i = 0; i < selfTransactions; i++) {
+    const tx = await deployer.sendTransaction({
+      to: deployer.address,
+      value: ethers.utils.parseEther('0'),
+      gasPrice: gasPrice ? gasPrice : 0
+    });
+    await tx.wait();
+    console.log(`Self transaction with nonce: ${i} complete`);
+  }
+}
 
-  const [deployer] = await ethers.getSigners();
+function getContractNames(version: string, chainId: number) {
+  if (version === 'v1') {
+    return [
+      'WalletSimple',
+      'WalletFactory',
+      'Forwarder',
+      'ForwarderFactory',
+      'contracts/WalletSimple.sol:WalletSimple'
```

Comment:
we should return coin specific contracts here

Reply:
Done.